Data/IntSet/EnumSet2.hs

```diff
@@ -1,15 +1,20 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 module Data.IntSet.EnumSet2 where
 
+#if MIN_VERSION_base(4,9,0)
+import Data.Semigroup
+#endif
 import qualified Data.IntSet as S
 import qualified Data.List as L (map)
 import Data.Monoid(Monoid(..))
 
 newtype EnumSet e = EnumSet {unEnumSet :: S.IntSet}
-  deriving (Eq,Ord,Read,Show)
-
-instance Monoid (EnumSet e) where
-  mempty = EnumSet mempty
-  EnumSet x `mappend` EnumSet y = EnumSet (x `mappend` y)
+  deriving (Eq,Ord,Read,Show,Monoid
+#if MIN_VERSION_base(4,9,0)
```

Comment:
this ones risky, as it may result in 3-package interactions if the package `EnumSet` doesn't provide Semigroup instances in whenever the CPP expects it.

If this happens, Hackage Trustees will be forced to make rather drastic revisions to add overconstraining version bounds to avoid the respective configurations to become available to the solver... :-/